spec/project_spec.rb

```diff
@@ -834,7 +834,8 @@ class InvalidErrorHandler; end
       expect(spy_logger).to have_received(:log).once.with(Logger::INFO, "Feature 'multi_variate_feature' is enabled for user 'test_user'.")
     end
 
-    it 'should return false, if the user is bucketed into a feature experiment but the featureEnabled property is false' do
+    it 'should return false and send impression if the user is bucketed into a feature experiment but the featureEnabled property is false' do
+      allow(project_instance.event_dispatcher).to receive(:dispatch_event).with(instance_of(Optimizely::Event))
```

Comment:
Can we look at content of dispatch event itself?

Reply:
I did try and it was giving me a lot of headache lol. The compat suite already checks the payload  so I'd say we merge this in!

Reply:
Go for it.